### ISSUE TITLE
[SF] Icon: click event cannot be triggered by "Enter"/"Space" key with screen reader on

### ISSUE THREAD

**Issue by summer2200** [SAP/ui5-webcomponents-react#7453]
### Describe the bug

This issue happens in Windows OS.

After turning on screen reader, tab to focus on the Icon, press enter or space key, the click event cannot be triggered.

This issue was found by our customer, please take it as a high priority. Thanks!

### Isolated Example

https://stackblitz.com/edit/github-ysh8324t?file=src%2FApp.tsx,src%2Fmain.tsx

### Reproduction steps

1. Open the isolated example in a Window OS browser
2. Tab to focus on the icon, press enter or space key, popover can be opened, pressing esc key, popover can be closed.
3. Turn on screen reader.
4. Do the same in step 2.

### Expected Behaviour

onClick event should be triggered when screen reader on in Windows OS.

### Screenshots or Videos

https://github.com/user-attachments/assets/810e515d-3715-4fd9-8428-a7dbb84423fe

### UI5 Web Components for React Version

2.10.0

### UI5 Web Components Version

2.10.0

### Browser

Chrome

### Operating System

Windows 10 Enterprise

### Relevant log output

```shell

```

### Declaration

- [x] I’m not disclosing any internal or sensitive information.

**Comment by Lukas742** [comment]
Thanks for reporting! I'll forward this issue to our UI5 Web Components Colleagues as the affected component is developed in their repository.
